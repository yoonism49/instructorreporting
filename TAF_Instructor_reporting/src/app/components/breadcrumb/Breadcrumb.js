// @flow weak

/* eslint consistent-return: 0 */
import React      from 'react';
import PropTypes  from 'prop-types';

const Breadcrumb = ({
  path
}) => (
  <ul className="breadcrumb">
  {
    path.length === 0 &&
    <div></div>
  }
  {
    path.length > 0 &&
    path.map(
      (view, viewIndex) => {
        if (viewIndex === 0) {
          return (
            <li key={viewIndex}>
              <a href="#">
                <i className="fa fa-home"></i>
                  &nbsp;
                  {view}
              </a>
            </li>
          );
        }
        if (viewIndex < path.length - 1) {
          return (
            <li
              key={viewIndex}
              className="active">
              {view}
            </li>
          );
        }
        if (viewIndex === path.length - 1) {
          return (
            <li key={viewIndex}
              className="active">
              {view}
            </li>
          );
        }
      }
    )
  }
  </ul>
);

Breadcrumb.propTypes = {
  path: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired
};

export default Breadcrumb;
