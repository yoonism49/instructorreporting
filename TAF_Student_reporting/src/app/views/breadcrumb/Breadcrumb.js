// @flow weak

import React, {
  PureComponent
}                         from 'react';
import PropTypes          from 'prop-types';
import {
  AnimatedView,
  Panel,
  Breadcrumb as BreadcrumbComponent
}                         from '../../components';
import Highlight          from 'react-highlight';


class Breadcrumb extends PureComponent {
  static propTypes= {
    actions: PropTypes.shape({
      enterBreadcrumb: PropTypes.func.isRequired,
      leaveBreadcrumb: PropTypes.func.isRequired
    })
  };

  state = { path: ['home', 'breadcrumb'] };

  componentWillMount() {
    const { actions: { enterBreadcrumb } } = this.props;
    enterBreadcrumb();
  }

  componentWillUnmount() {
    const { actions: { leaveBreadcrumb } } = this.props;
    leaveBreadcrumb();
  }

  render() {
    const { path } = this.state;

    const source = `
      // import
      import { Breadcrumb } from './_SOMEWHERE_/components';

      // path (in state for example):
      state = {
        path: ['home', 'breadcrumb']
      };

      // in render():
      <BreadcrumbComponent
        path={this.state.path}
      />
      `;

    return(
      <AnimatedView>
        {/* preview: */}
        <div className="row">
          <div className="col-xs-12">
            <Panel
              title="Breadcrumb"
              hasTitle={true}
              bodyBackGndColor={'#F4F5F6'}>
              <div className="row">
                <div className="col-xs-12">
                  <BreadcrumbComponent
                    path={path}
                  />
                </div>
              </div>
            </Panel>
          </div>
        </div>
        {/* source: */}
        <div className="row">
          <div className="col-xs-12">
            <Panel
              title="Source"
              hasTitle={true}>
              <Highlight className="javascript">
                {source}
              </Highlight>
            </Panel>
          </div>
        </div>
      </AnimatedView>
    );
  }
}

export default Breadcrumb;
