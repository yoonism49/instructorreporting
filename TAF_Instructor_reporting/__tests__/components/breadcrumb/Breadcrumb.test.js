'use strict';

import React          from 'react';
import Breadcrumb     from '../../../src/app/components/breadcrumb/Breadcrumb';
import renderer       from 'react-test-renderer'; // needed both for snpashot testing but also to prevent errors from enzyme


describe('Breadcrumb component', () => {
  const mockPath = ['path1', 'subpath1'];
  it('renders as expected', () => {
    const component = renderer.create(
      <div>
        <Breadcrumb path={mockPath} />
      </div>
    ).toJSON();
    expect(component).toMatchSnapshot();
  });
});
