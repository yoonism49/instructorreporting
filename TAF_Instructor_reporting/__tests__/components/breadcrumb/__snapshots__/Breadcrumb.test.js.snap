// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb component renders as expected 1`] = `
<div>
  <ul
    className="breadcrumb"
  >
    <li>
      <a
        href="#"
      >
        <i
          className="fa fa-home"
        />
        Â 
        path1
      </a>
    </li>
    <li
      className="active"
    >
      subpath1
    </li>
  </ul>
</div>
`;
